odxtools/communicationparameter.py

```diff
@@ -1,44 +1,25 @@
 # SPDX-License-Identifier: MIT
 # Copyright (c) 2022 MBition GmbH
 
-from typing import Optional, List, TYPE_CHECKING
+from typing import Optional, Any, Union, List, TYPE_CHECKING
 import warnings
 
 from odxtools.exceptions import OdxWarning
 
 from .odxlink import OdxLinkDatabase, OdxLinkRef, OdxDocFragment
 from .utils import read_description_from_odx
-from .comparam_subset import BaseComparam, Comparam, ComplexComparam, read_complex_value_from_odx
-
-
-def _get_comparam_value(comparam: BaseComparam, value):
-    if not comparam:
-        return value
-    if value is None:
-        value = comparam.physical_default_value
-
-    if isinstance(comparam, Comparam):
-        return comparam.dop.physical_type.base_data_type.from_string(value)
```

Comment:
@andlaus original code here did take into consideration the dop physical data type

Reply:
yes, but that was a mightily confusing layer above `CommunicationParameterRef.value`. I suppose that in the future, this should be moved to `CommunicationParameterRef`. (which is a bit more work than it first seems, because of the writing stuff...)